Sources/Algorithms/Cycle.swift

```diff
@@ -56,6 +56,47 @@ extension Cycle: Sequence {
 
 extension Cycle: LazySequenceProtocol where Base: LazySequenceProtocol {}
 
+
+/// A collection wrapper that repeats the elements of a base collection for a finite number of times.
+public struct FiniteCycle<Base: Collection> {
+  /// The collection to repeat.
+  public let base: Base
```

Comment:
Let's make this `internal`, as well as `times`.

This is because equality (we could conform this type to `Equatable`, but it doesn't have to be here in this PR) should be based on elementwise comparisons, and `[1, 2, 3].cycled(times: 6)` should be equivalent to `[1, 2, 3, 1, 2, 3].cycled(times: 3)`. Having `base` and `times` publicly accessible would raise the question of whether these properties are "salient" ones for the purposes of considering equivalence.

Reply:
Should I also change `Cycle.base` to internal as well? This was why I used public here.